source/slang/slang-shader-cache-index.cpp

```diff
@@ -0,0 +1,123 @@
+#include "slang-shader-cache-index.h"
+
+#include "slang-hash-utils.h"
+#include "../core/slang-io.h"
+
+namespace Slang
+{
+
+// Load a previous cache index saved to disk. If not found, create a new cache index
+// and save it to disk as filename.
+SlangResult ShaderCacheIndex::loadCacheIndexFromFile(String filename)
+{
+    ComPtr<ISlangBlob> indexBlob;
+    if (SLANG_FAILED(shaderCacheFileSystem->loadFile(filename.getBuffer(), indexBlob.writeRef())))// Should use the mutable system if available
+    {
+        // Cache index not found, so we'll create and save a new one.
+        if (mutableShaderCacheFileSystem)
+        {
+            SLANG_RETURN_ON_FAIL(mutableShaderCacheFileSystem->saveFile(filename.getBuffer(), nullptr, 0));
+            indexFilename = filename;
+            return SLANG_OK;
+        }
+    }
+
+    String indexString;
+    File::readAllText(filename, indexString);
```

Comment:
You can use `StringUtil::calcLines` to get your a `List<UnownedStringSlice> lines` where each element represents a line.

Then you can just iterate through the lines:
```
for (auto line : lines)
{
    List<UnownedStringSlice> words;
    StringUtil::split(line, ' ', words); // words will contain two elements, one for each hash.
    auto dependencyDiggest = stringToDigest(words[0]);
    auto astDigest = stringToDigest(words[1]);
    ...
}
```